src/libs/antares/study/parts/hydro/hydromaxtimeseriesreader.cpp

```diff
@@ -0,0 +1,214 @@
+
+    ret = maxDailyGenEnergy.saveToCSVFile(filePath, 2) && ret;
+
+    return ret;
+}
+
+bool HydroMaxTimeSeriesReader::SaveMaxPumpingEnergy(const AnyString& folder, Area& area)
+{
+    YString filePath;
+    auto& maxDailyPumpEnergy = area.hydro.maxDailyPumpEnergy;
+    bool ret = true;
+
+    maxDailyPumpEnergy.reset(1, DAYS_PER_YEAR, true);
+    maxDailyPumpEnergy.fillColumn(0, 24.);
+
+    for (uint day = 0; day < DAYS_PER_YEAR; ++day)
+        maxDailyPumpEnergy[0][day] = dailyMaxPumpAndGen[pumpMaxE][day];
+
+    filePath.clear() << folder << SEP << "common" << SEP << "capacity" << SEP
+                     << "maxDailyPumpEnergy_" << area.id << ".txt";
+    ret = maxDailyPumpEnergy.saveToCSVFile(filePath, 2) && ret;
+
+    return ret;
+}
+
+bool HydroMaxTimeSeriesReader::SaveDailyMaxPowerAsHourly(const AnyString& folder, Area& area)
+{
+    bool ret = true;
+
+    ret = SaveDailyMaxGenPowerAsHourly(folder, area) && ret;
+    ret = SaveDailyMaxPumpPowerAsHourly(folder, area) && ret;
+    area.hydro.series->setCountVariable();
+
+    return ret;
+}
+
+bool HydroMaxTimeSeriesReader::SaveDailyMaxGenPowerAsHourly(const AnyString& folder, Area& area)
+{
+    bool ret = true;
+    YString filePath;
+
+    auto& maxHourlyGenPower = area.hydro.series->maxHourlyGenPower;
+    auto& dailyMaxGen = dailyMaxPumpAndGen[genMaxP];
+
+    maxHourlyGenPower.reset(1, HOURS_PER_YEAR);
+
+    TransferDailyMaxPowerAsHourly(maxHourlyGenPower[0], dailyMaxGen);
+
+    filePath.clear() << folder << SEP << "series" << SEP << area.id << SEP << "maxHourlyGenPower."
+                     << "txt";
+    ret = maxHourlyGenPower.saveToCSVFile(filePath, 2) && ret;
+
+    return ret;
+}
+
+bool HydroMaxTimeSeriesReader::SaveDailyMaxPumpPowerAsHourly(const AnyString& folder, Area& area)
+{
+    bool ret = true;
+    YString filePath;
+
+    auto& maxHourlyPumpPower = area.hydro.series->maxHourlyPumpPower;
+    auto& dailyMaxPump = dailyMaxPumpAndGen[pumpMaxP];
+
+    maxHourlyPumpPower.reset(1, HOURS_PER_YEAR);
+
+    TransferDailyMaxPowerAsHourly(maxHourlyPumpPower[0], dailyMaxPump);
+
+    filePath.clear() << folder << SEP << "series" << SEP << area.id << SEP << "maxHourlyPumpPower."
+                     << "txt";
+    ret = maxHourlyPumpPower.saveToCSVFile(filePath, 2) && ret;
+
+    return ret;
+}
+
+bool HydroMaxTimeSeriesReader::operator()(const AnyString& folder, Area& area)
+{
+    bool ret = true;
+
+    ret = LoadDailyMaxPowersAndEnergies(folder, area) && ret;
+    ret = SaveDailyMaxEnergy(folder, area) && ret;
+    ret = SaveDailyMaxPowerAsHourly(folder, area) && ret;
+
+    return ret;
+}
+
+void TransferDailyMaxPowerAsHourly(Matrix<double, int32_t>::ColumnType& hourlyColumn,
+                                    const Matrix<double>::ColumnType& dailyColumn)
+{
+    uint hours = 0;
+    uint days = 0;
+
+    while (hours < HOURS_PER_YEAR && days < DAYS_PER_YEAR)
+    {
+        for (uint i = 0; i < 24; ++i)
```

Comment:
```suggestion
        for (uint i = 0; i < HOURS_PER_DAY; ++i)
```
?

Reply:
I've implemented that macro earlier today when i was refactoring AreaScratchpad class, but I did not pushed it. Regarding AreaScratchpad class will you be able to give me your comment on this?
https://github.com/AntaresSimulatorTeam/Antares_Simulator/pull/1470#discussion_r1333034108